### DISCUSSION TITLE
classes/class_physicsshapequeryparameters3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#285] [User-contributed notes]
# classes/class_physicsshapequeryparameters3d

Inherits: RefCounted< Object Provides parameters for PhysicsDirectSpaceState3D.intersect_shape. Description: By changing various properties of this object, such as the shape, you can configure the ...

https://docs.godotengine.org/en/stable/classes/class_physicsshapequeryparameters3d.html

**Comment by miasmile111** [comment]
Heads up, the example for the 'shape_rid' property seems to be outdated, the line:
```
var shape_rid = PhysicsServer3D.shape_create(PhysicsServer3D.SHAPE_SPHERE)
```
seems to have been updated to be
```
var shape_rid = PhysicsServer3D.sphere_shape_create()
```
There may be other such inconsistencies in the syntax! :)

**Reply by jlitowitz** [reply]
In addition, in the C# example, `params` is a reserved keyword so that's invalid syntax.  I modified my local example to this, and at least now I can compile:

``` cs
		Rid shapeRid = PhysicsServer3D.SphereShapeCreate();
		float radius = 2.0f;
		PhysicsServer3D.ShapeSetData(shapeRid, radius);

		var queryParams = new PhysicsShapeQueryParameters3D();
		queryParams.ShapeRid = shapeRid;

		// Execute physics queries here...

		// Release the shape when done with physics queries.
		PhysicsServer3D.FreeRid(shapeRid);

```
